### PR TITLE
Remove `test/integration_tests/test_sampler.py`

### LINKED ISSUE #4030: Remove `test/integration_tests/test_sampler.py`
### Motivation

`test/integration_tests/test_sampler.py` has common tests functions for samplers in integration modules such as `CmaEsSampler` and `SkoptSampler`. However, it seems to be outdated as follows:
- most of the test cases are covered by the other test case, and
- it does not covers all samplers in the integration module like `BoTorchSampler`

### Description

`test/integration_tests/test_sampler.py` has three test functions. `test_suggested_value` and `test_sample_independent` can be simply removed. Please re-implement `test_warn_independent_sampling` to `tests/integration_tests/test_cma.py` and `tests/integration_tests/test_skopt.py` based on the [test function of `TPESampler`](https://github.com/optuna/optuna/blob/ceac5f13aa7c9ec9f5a7b08d59dec4c3bdbb2b2a/tests/samplers_tests/tpe_tests/test_sampler.py#L44-L62).

- `test_suggested_value` checks the range of suggested values. It is covered by [`test_float`](https://github.com/optuna/optuna/blob/58434f930c606f539ff75d0a1cd96c1f82362e9e/tests/samplers_tests/test_samplers.py#L261-L264), [`test_int`](https://github.com/optuna/optuna/blob/58434f930c606f539ff75d0a1cd96c1f82362e9e/tests/samplers_tests/test_samplers.py#L301) and [`test_categorical`](https://github.com/optuna/optuna/blob/58434f930c606f539ff75d0a1cd96c1f82362e9e/tests/samplers_tests/test_samplers.py#L320) in `tests/samplers_tests/test_sampler.py` .
- `test_sample_independent` checks if `sample_independent` method is called as expected. It seems to be a test of `Trial`, and it is actually covered by [`tests/trial_tests/test_trial.py`](https://github.com/optuna/optuna/blob/2c09c5d689411aae19a613970ade3a1b54b01ddb/tests/trial_tests/test_trial.py#L392-L427).
- `test_warn_independent_sampling` checks if `sampler._log_independent_sampling` method is called as expected. Not all samplers implement `sample_relative` method, and it should be tested in [individual test files like `TPESampler`](https://github.com/optuna/optuna/blob/ceac5f13aa7c9ec9f5a7b08d59dec4c3bdbb2b2a/tests/samplers_tests/tpe_tests/test_sampler.py#L44-L62)

### Alternatives (optional)

_No response_

### Additional context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Resolve #4030

## Description of the changes
Remove test cases in `test_suggested_value()` and `test_sample_independent()` since these are already covered in `tests/samplers_tests/test_sampler.py` and `/tests/trial_tests/test_trial.py`, respectively.
Re-implemente test cases in `test_warn_independent_sampling()`, which checks if sampler._log_independent_sampling method is called as expected, to `tests/integration_tests/test_cma.py` and `tests/integration_tests/test_skopt.py`, respectively. New implementataion is based on `test_warn_independent_sampling()` in `tests/samplers_tests/tpe_tests/test_sampler.py`.

### CODE REVIEW

File: tests/integration_tests/test_cma.py [hunk 1]
```diff
@@ -70,6 +72,29 @@ def test_init_default_values() -> None:
...
+        optuna.logging.set_verbosity(optuna.logging.WARNING)
+
+        sampler = optuna.integration.PyCmaSampler(
+            warn_independent_sampling=True, n_startup_trials=0
+        )
+        study = optuna.create_study(sampler=sampler)
+        study.optimize(objective, n_trials=10)
+
+        _, err = capsys.readouterr()
+        assert err
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_cma.py:96]: [Note] This is based on the test case of `TPESampler`.
https://github.com/optuna/optuna/blob/ceac5f13aa7c9ec9f5a7b08d59dec4c3bdbb2b2a/tests/samplers_tests/tpe_tests/test_sampler.py#L44-L62

File: tests/integration_tests/test_skopt.py [hunk 2]
```diff
@@ -136,6 +138,27 @@ def test_is_compatible() -> None:
...
+    optuna.logging._reset_library_root_logger()
+    optuna.logging.enable_default_handler()
+    optuna.logging.set_verbosity(optuna.logging.WARNING)
+
+    sampler = optuna.integration.SkoptSampler(warn_independent_sampling=True, n_startup_trials=0)
+    study = optuna.create_study(sampler=sampler)
+    study.optimize(objective, n_trials=10)
+
+    _, err = capsys.readouterr()
+    assert err
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_skopt.py:159]: ditto.

File: tests/integration_tests/test_sampler.py [hunk 3]
```diff
@@ -1,170 +0,0 @@
...
-            assert distribution._contains(param_value_in_internal_repr)
-
-    # direction='maximize'
-    study = optuna.create_study(sampler=sampler, direction="maximize")
-    study.optimize(_objective, n_trials=10, catch=())
-    for trial in study.trials:
-        for param_name, param_value in trial.params.items():
-            distribution = trial.distributions[param_name]
-            param_value_in_internal_repr = distribution.to_internal_repr(param_value)
-            assert distribution._contains(param_value_in_internal_repr)
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_sampler.py:47]: [Note] I found that `test_float` in tests/samlers_test/test_samplers.py checks only 
`minimize`, but I don't think we need to keep this test case only for `CmaSampler` and `SkoptSampler`.

https://github.com/optuna/optuna/blob/58434f930c606f539ff75d0a1cd96c1f82362e9e/tests/samplers_tests/test_samplers.py#L261-L264

**Review by toshihikoyanase — APPROVED** [review]
Looks good to me. Thank you for your PR!

**Review by gen740 — APPROVED** [review]
LGTM!

**Pull request merged**